dao/src/main/java/cz.muni.fi.pa165.soccermanager/entity/League.java

```diff
@@ -1,4 +1,90 @@
 package cz.muni.fi.pa165.soccermanager.entity;
 
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.List;
+
 public class League {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+    private String name;
+    private String country;
+
+
+    //List<Match> matches;
+
+    ArrayList<Match> matches = new ArrayList<Match>();
```

Comment:
change the variable type to List<Match> matches = new ArrayList<Match>();
It is better to work with interfaces
https://stackoverflow.com/questions/2279030/type-list-vs-type-arraylist-in-java

Reply:
also in nowdays versions of java we dont need to write the generic type Match in the constructor - java is able to find out from the context. simply:
List<Match> matches = new ArrayList<>();